app/classifier/drawing-tools/freehand-segment-shape.jsx

```diff
@@ -0,0 +1,250 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import DrawingToolRoot from './root';
+import DeleteButton from './delete-button';
+import { createPathFromCoords, filterDupeCoords, roundCoords } from './freehand-helpers';
+
+const BUFFER = 16;
+const DELETE_BUTTON_WIDTH = 8;
+const FINISHER_RADIUS = 8;
+const POINT_RADIUS = 4;
+
+class FreehandSegmentShapeTool extends React.Component {
+  static initCoords = null;
+
+  static defaultValues() {
+    return {
+      points: [],
+      _inProgress: false,
+      _currentlyDrawing: false
+    };
+  }
+
+  static initStart(coords, mark) {
+    mark.points.push(roundCoords(coords));
+    mark._inProgress = true;
+    mark._currentlyDrawing = true;
+    return mark;
+  }
```

Comment:
I'm not sure, but I think this should return a new object, rather than mutating the mark directly. See the Fan tool, for example. https://github.com/zooniverse/Panoptes-Front-End/blob/07e975ef1f7a3918c7a966e3659974ea89d97327/app/classifier/drawing-tools/components/Fan/fan.jsx#L24-L26

Reply:
Ahh, this makes total sense. Thanks for pointing this out. Was a bit confused initially at the best way of returning the mark.

Reply:
The Readme isn't very clear. I only figured it out by reading the drawing task code to see how these functions are used.